tests/test_unauthenticated_reddit.py

```diff
@@ -332,3 +336,11 @@ def test_store_lazy_json_result(self):
 
     def test_user_agent(self):
         self.assertWarnings(UserWarning, Reddit, 'robot agent')
+        google_app_engine = "Google App Engine v2.6"
+        old_server_software, os.environ['SERVER_SOFTWARE'] = \
+            os.environ.get('SERVER_SOFTWARE'), google_app_engine
+        self.assertIn(google_app_engine,
+                      Reddit('robot engine').http.headers['User-Agent'])
+        del os.environ['SERVER_SOFTWARE']
```

Comment:
Can you use a finally block here? If the assert fails, the env won't be restored.